lib/uiautomator2.js

```diff
@@ -35,33 +39,31 @@ class UiAutomator2Server {
    * @param {number} installTimeout - Installation timeout
    */
   async installServerApk (installTimeout = SERVER_INSTALL_RETRIES * 1000) {
-    const serverApkInfo = await this.adb.getApkInfo(apkPath);
-    const apkPackage = serverApkInfo.name;
-    if (!apkPackage) {
-      throw new Error(`Cannot read manifest information from '${apkPath}'. Does the package exist and is accessible?`);
-    }
-    // appending .test to apkPackage name to get test apk package name
-    const testApkPackage = `${apkPackage}.test`;
-    let isApkInstalled = await this.adb.isAppInstalled(apkPackage);
-    let isTestApkInstalled = await this.adb.isAppInstalled(testApkPackage);
-    if (isApkInstalled || isTestApkInstalled) {
-      const testPackageInfo = await this.adb.getPackageInfo(apkPackage);
-      //check server apk versionName
-      const apkVersion = serverApkInfo.versionName;
-      const pkgVersion = testPackageInfo.versionName;
-      if (!apkVersion || !pkgVersion || apkVersion !== pkgVersion) {
-        logger.debug(`Server installed but version ${pkgVersion} instead of ${apkVersion}. Re-installing`);
-        isApkInstalled = isTestApkInstalled = false;
-        await this.uninstallServerAndTest(apkPackage, testApkPackage);
-      }
-    }
-
-    if (!isApkInstalled) {
-      await this.signAndInstall(apkPath, apkPackage, installTimeout);
-    }
-    if (!isTestApkInstalled) {
-      await this.signAndInstall(testApkPath, testApkPackage, installTimeout, true);
+    const packagesInfo = [
+      {
+        appPath: apkPath,
+        appId: SERVER_PACKAGE_ID,
+      },
+      {
+        appPath: testApkPath,
+        appId: SERVER_TEST_PACKAGE_ID,
+      },
+    ];
+    if (await B.reduce(packagesInfo,
+                       async (accumulator, pkgInfo) => (await this.checkAndSignCert(pkgInfo.appPath, pkgInfo.appId)) || accumulator,
```

Comment:
maybe breaking out the accumulator method would make this more readable?

Reply:
accumulator is just a value. The idea is that if any of binaries is unsigned then we need to resign them and uninstall from the device

Reply:
makes sense. my only issue now is it looks ugly :-)